docs/adr/0002-delivery-channel-creation-and-integration.md

```diff
@@ -0,0 +1,81 @@
+
+## Decision
+```
+                 +------------+
+                 |            |
+                 | Management |
+                 | API        |
+                 |            |
+                 |            |
+                 ++--+------+-+
+                  ^  ^      ^
+                  |  |      |
+         +--------+  |      +--------------+
+         |           |                     |
+         v           v                     v
++--------+-+      +--+-------+           +-+--------+
+|          |      |          |           |          |
+| Delivery |      | Delivery |           | Delivery |
+| Channel  |      | Channel  | +-+  +-+  | Channel  |
+| 1        |      | 2        |           | N        |
+|          |      |          |           |          |
+|          |      |          |           |          |
++----------+      +----------+           +----------+
+```
+
+The management api should have knowledge of what delivery channels are available and where certain notification types should be sent. As we send notifications through on a per user per tenant basis, user settings for which communication type(s) a user has allowed is read and a notification is sent to the available delivery channel(s). 
+
+### User notifications and user settings
+
+A user should be able to choose what kind of communication types they receive based on the notification type. There are two ways of approaching this, either:
+1. Send different types of notifications through all channels that are provided by the consumer of this service, and leave the communication type and user configuration for the consumer of the Marain.UserNotifications service. (ie, the consumer of the Marain.UserNotifications service should specify which channels they want to send notifications via an api call when a notification is created)
+2. Store user preferences somewhere that is accessible to the management api, then when a new type of notifications comes through, check what type of communications types  the user has agreed to use and then send a message through the delivery channel .
+
+The latter option was chosen as it provides a user the ability to choose what notifications they want out of the box, if the former is chosen, all users/businesses integrating with this service will then need to write their own notifications user management service.
+
+User settings for which notifications will go to what communication channels for each user will be stored inside a User Notifications Settings table which will be accessed by the Management Api. Configuration in this table will include the Notification Type and the appropriate communication types associated to that notification type. This table will have an associated crud api that will allow a user to view/change the stored settings.
```

Comment:
Well, that would depend on how the end user would want to configure. The idea is that the api will have a facade api layer between the `Marain.UserNotifications` service and the UI.

The intention is that the api is not directly accessible from any end-users machine.

Reply:
>The intention is that the api is not directly accessible from any end-users machine.

OK, that sounds good. But then I think the language needs clarification in this last sentence:

>This table will have an associated crud api that will allow a user to view/change the stored settings.

From what you've said, I think this API enables an application developer to provide the necessary elements (facade API and UI, as you say) to enable the user to view and change the settings. As early, I think this might be a case of being a bit broad with the term "user".

Reply:
Agreed. I've updated this and have also changed the beginning of this paragraph as it was confusing.